engine/api/application_vulnerability.go

```diff
@@ -28,25 +29,30 @@ func (api *API) postVulnerabilityHandler() service.Handler {
 			return sdk.WrapError(err, "Unable to read body")
 		}
 
-		app, errL := application.LoadByName(api.mustDB(), key, appName)
-		if errL != nil {
-			return sdk.WrapError(errL, "Unable to load application")
+		proj, err := project.Load(api.mustDB(), projectKey)
```

Comment:
I think it's a bit sad to load the project to get its ID, whereas the projectKey is an identifier.
we should consider `application.LoadByProjectKeyAndName`